app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorTest.kt

```diff
@@ -69,6 +69,47 @@ class UriUtilsFilenameExtractorTest {
         assertEquals("realFilename.jpg", extractionResult.filename)
     }
 
+    @Test
+    fun whenUrlContainsFilenameButContainsPathSegmentsWhichLookLikeAFilenameThenFilenameShouldBeExtracted() {
+        val url = "https://foo.example.com/path/dotted.path/b/b1/realFilename.jpg"
+        val mimeType: String? = null
+        val contentDisposition: String? = null
+
+        val extractionResult = testee.extract(buildPendingDownload(url, contentDisposition, mimeType))
+        assertTrue(extractionResult is FilenameExtractor.FilenameExtractionResult.Extracted)
+
+        extractionResult as FilenameExtractor.FilenameExtractionResult.Extracted
+        assertEquals("realFilename.jpg", extractionResult.filename)
+    }
+
+    @Test
+    @SdkSuppress(minSdkVersion = 22)
```

Comment:
not sure if this is necessary, can you double check?

Reply:
From what I gather, `@SdkSuppress(minSdkVersion = 22)` is used in the [existing MIME type tests](https://github.com/duckduckgo/Android/blob/784c87beef30a351436f8a5300950b6f7081c77e/app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorTest.kt#L167-L195) because in SDK version 21 and below, `image/jpeg` defaults to a `.jpeg` file extension, and SDK 22 and above defaults to a `.jpg` file extension (dropped the `e`). So using it here just ensures that the test doesn't fail if/when we execute it against a lower SDK target.

In the case of this test case, I _think_ it's required to satisfy the assertion, but I will double-check.

Reply:
Thanks for the explanation, I was not aware of that 👍